### PR TITLE
detect/bytetest: fix incorrect NULL check for offset pointer

### PR DESCRIPTION
**Pull request by afonot**
DetectBytetestParse() checked data_offset for NULL instead of the caller-supplied offset pointer. When NULL was passed for offset, this led to a NULL-pointer dereference and a crash.

Replaced the condition with the correct `if (offset == NULL)`.

Bug 7767

Found with SVACE static analyzer

## Contribution style:
- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html

## Our Contribution agreements:
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)

## Changes (if applicable):
- [x] I have created a ticket at
      https://redmine.openinfosecfoundation.org/projects/suricata/issues

Link to ticket: https://redmine.openinfosecfoundation.org/issues/7767

### CODE REVIEW

File: src/detect-bytetest.c [hunk 1]
```diff
@@ -468,7 +468,7 @@ static DetectBytetestData *DetectBytetestParse(
         memmove(data_offset, str_ptr, end_ptr - str_ptr);
         data_offset[end_ptr-str_ptr] = '\0';
         if (data_offset[0] != '-' && isalpha((unsigned char)data_offset[0])) {
-            if (data_offset == NULL) {
+            if (offset == NULL) {
```
Comment (catenacyber) [inline src/detect-bytetest.c]: Indeed the previous check looked clearly wrong and impossible (as we checked `if (data_offset)` in the surrounding block a few lines above)

Would you have a test case for this code path ?

Comment (catenacyber) [reply src/detect-bytetest.c]: The fix does not seem right : `offset` is never NULL from the caller

Comment (afonot) [reply src/detect-bytetest.c]: > The fix does not seem right : `offset` is never NULL from the caller

You're right — offset can't be NULL. Do you think it's appropriate to remove the check entirely in that case?

Comment (catenacyber) [reply src/detect-bytetest.c]: > Do you think it's appropriate to remove the check entirely in that case?

Sounds right :-)

Need to check that unit tests are still ok (as they use NULL)
If they are not, the check `if (offset == NULL)` can be surrounded by #ifdef UNITTESTS

Comment (afonot) [reply src/detect-bytetest.c]: > Need to check that unit tests are still ok (as they use NULL) If they are not, the check `if (offset == NULL)` can be surrounded by #ifdef UNITTESTS

Done.
In the unit tests, offset is NULL, but those tests pass the value parameter as NULL, which causes the
function to return before offset is dereferenced.

**Comment by catenacyber** [comment]
(otherwise, looks good)
